vernac/g_vernac.mlg

```diff
@@ -693,18 +693,20 @@ GRAMMAR EXTEND Gram
           LIST1 [ v=strategy_level; "["; q=LIST1 smart_global; "]" -> { (v,q) } ] ->
             { VernacSetStrategy l }
       (* Canonical structure *)
-      | IDENT "Canonical"; IDENT "Structure"; qid = global ->
-          { VernacCanonical CAst.(make ~loc @@ AN qid) }
-      | IDENT "Canonical"; IDENT "Structure"; ntn = by_notation ->
+      | IDENT "Canonical"; OPT [ IDENT "Structure" -> {()} ]; qid = global; ud = OPT [ u = OPT univ_decl; d = def_body -> { (u,d) } ] ->
```

Comment:
Why is `-> {()}` needed?

Reply:
Because `OPT IDENT "Structure "` does not parse and if you put `[` then it wants an `->`... Or at least this is my understanding of the errors I hot

Reply:
Ok, thanks!